### PR TITLE
PXD-1315 Fix linting and test errors in console

### PR DESCRIPTION
**Pull request by abgeorge7**
Fixed all our warnings/errors in the console to make linting and testing easier. These things are mostly default props, multiple components in one file, nested ternaries, default exports, etc.

Also put a test in that will fail if there are any console warnings, so that in the future we won't push code that causes these warnings.

Would appreciate a lot of eyes on this, I refactored a lot of code I'm not super familiar with so I want to be sure nothing breaks.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -111,6 +111,7 @@
     "verbose": true,
     "automock": false,
     "setupFiles": [
+      "./src/shim.js",
```
Comment (frickjack) [inline package.json]: Can we just merge this into `setupJest.js`?

File: src/CoreMetadata/page.jsx [hunk 2]
```diff
@@ -20,7 +20,7 @@ class CoreMetadataPage extends Component {
   render() {
     return (
       <div>
-        <BackLink url='/files' label='Back to data overview' />
+        <BackLink url="/files" label="Back to data overview" />
```
Comment (frickjack) [inline src/CoreMetadata/page.jsx]: We're supposed to use single-quotes in javascript, and double quotes in jsx?  Ugh.

File: src/DataDictionary/DataDictionary.jsx [hunk 3]
```diff
@@ -90,9 +90,10 @@ export function category2NodeList(dictionary) {
   )
     .reduce(
       (lookup, node) => {
-        if (!lookup[node.category]) { lookup[node.category] = []; }
-        lookup[node.category].push(node);
-        return lookup;
+        let category = lookup[node.category];
+        if (!lookup[node.category]) { category = []; }
+        category.push(node);
+        return Object.assign({}, lookup, { [node.category]: category });
```
Comment (frickjack) [inline src/DataDictionary/DataDictionary.jsx]: Why are we doing this?
Every time through the reduce loop we're making a new copy of the accumulator dictionary - so we go from O(n) to O(n^2) ... there's no reason to do that

Comment (abgeorge7) [reply src/DataDictionary/DataDictionary.jsx]: Linting gives the error `Assignment to property of function parameter 'lookup'  no-param-reassign` so I am trying to eliminate that, I can disable this for `.reduce` though?

File: src/CoreMetadata/page.jsx (hunk 2 above)
Comment (abgeorge7) [reply src/CoreMetadata/page.jsx]: if there are conflicting styles I can try to standardize it

File: package.json (hunk 1 above)
Comment (abgeorge7) [reply package.json]: Unfortunately that doesn't get rid of the warning :( 
https://github.com/facebook/jest/issues/4545#issuecomment-332762365

File: src/CoreMetadata/page.jsx (hunk 2 above)
Comment (abgeorge7) [reply src/CoreMetadata/page.jsx]: Is there one we want? I can set that in ESLint so they are the same

File: src/DataDictionary/DataDictionary.jsx (hunk 3 above)
Comment (frickjack) [reply src/DataDictionary/DataDictionary.jsx]: Yeah - that would be great if you can eliminate that rule for .reduce :+1:

File: src/CoreMetadata/page.jsx (hunk 2 above)
Comment (abgeorge7) [reply src/CoreMetadata/page.jsx]: Standardized it to single - so there are a lot of changed files now

**Pull request merged**